doc/build/zephyr_cmake_package.rst

```diff
@@ -313,44 +313,70 @@ Such a CMakeLists.txt could look as:
+* **ZephyrAppConfiguration**: Applies only to the application you are currently building
 
-The Zephyr Build Configuration CMake package will be loaded in the Zephyr boilerplate code after
-initial properties and ``ZEPHYR_BASE`` has been defined, but before CMake code execution.
-This allows the Zephyr Build Configuration CMake package to setup or extend properties such as:
+They are similar to the per-user ``.zephyrrc`` file that can be used to set :ref:`env_vars`, but
+they set CMake variables instead. They also allow you to automatically share the build configuration
+among all users through the project repository. They also allow more advanced use cases, such as
+loading of additional CMake boilerplate code.
+
+The Zephyr Build Configuration CMake packages will be loaded in the Zephyr boilerplate code after
+initial properties and ``ZEPHYR_BASE`` has been defined, but before CMake code execution. The
+ZephyrBuildConfiguration is included first and ZephyrAppConfiguration afterwards. That means the
+application-specific package could override the workspace settings, if needed.
+This allows the Zephyr Build Configuration CMake packages to setup or extend properties such as:
 ``DTS_ROOT``, ``BOARD_ROOT``, ``TOOLCHAIN_ROOT`` / other toolchain setup, fixed overlays, and any
 other property that can be controlled. It also allows inclusion of additional boilerplate code.
 
-To provide a Zephyr Build Configuration CMake package, create ``ZephyrBuildConfig.cmake`` and place
-it in a Zephyr workspace top-level folder as:
+To provide a ZephyrBuildConfiguration or ZephyrAppConfiguration, create ``ZephyrBuildConfig.cmake``
+and/or ``ZephyrAppConfig.cmake`` respectively and place them in the appropriate location. The CMake
+``find_package`` mechanism will search for these files with the steps below. Other default
+CMake package search paths and hints are disabled and there is no version checking implemented for
+these packages. This also means that these packages cannot be installed in the CMake package
+registry. The search steps are:
+
+1. If ``ZephyrBuildConfiguration_ROOT``, or ``ZephyrAppConfiguration_ROOT`` respectively, is set,
+   search within this prefix path. If a matching file is found, execute this file. If no matching
+   file is found, go to step 2.
+2. Search within ``${ZEPHYR_BASE}/../*``, or ``${APPLICATION_SOURCE_DIR}`` respectively. If a
+   matching file is found, execute this file. If no matching file is found, abort the search.
+
+It is recommended to place the files in the default paths from step 2, but with the ``*_ROOT``
+variables you have the flexibility to place them anywhere. This is especially necessary for
+freestanding applications, for which the default path to ZephyrBuildConfiguration usually does not
+work. In this case the ``*_ROOT`` variables can be set on the CMake command line, **before**
+``find_package(Zephyr ...)`` or from a CMake cache initialization file with the ``-C`` command
+line option.
```

Comment:
CMake find package also allows those to be set as environment variable.
This may be useful for Zephyr build config, but is risky for app config.

So might be worth mentioning, but I leave the decision to the author / other reviewers opinions.

Reply:
Can certainly add this bit of information 👍